src/main/java/htsjdk/samtools/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,758 @@
+     * @throws IllegalStateException    if no base was added to the previous sequence or the writer is already closed.
+     * @throws IOException              if such exception is thrown when writing into the output resources.
+     */
+    public FastaReferenceWriter startSequence(final String sequenceName, final String description)
+            throws IOException {
+        return startSequence(sequenceName, description, defaultBasePerLine);
+    }
+
+    /**
+     * Starts the input of the bases of a new sequence.
+     * <p>
+     * This operation automatically closes the previous sequence base input if any.
+     * </p>
+     * <p>
+     * The sequence name cannot contain any blank characters (as determined by {@link Character#isWhitespace(char)}),
+     * control characters (as determined by {@link Character#isISOControl(char)}) or the the FASTA header star character
+     * {@value #HEADER_START_CHAR}. It cannot be the empty string either ("").
+     * </p>
+     * <p>
+     * The description cannot contain {@link Character#isISOControl(char)}. If set to {@code null} or the empty
+     * string ("") no description will be outputted.
+     * </p>
+     * <p>
+     * The input bases-per-line must be 1 or greater.
+     * </p>
+     * <p>
+     * This method cannot be called after the writer has been closed.
+     * </p>
+     * <p>
+     * It also will fail if no base was added to the previous sequence if any.
+     * </p>
+     *
+     * @param sequenceName the name of the new sequence.
+     * @param description  optional description for that sequence.
+     * @param basesPerLine number of bases per line for this sequence.
+     * @return this instance.
+     * @throws IllegalArgumentException if any argument does not comply with requirements listed above.
+     * @throws IllegalStateException    if no base was added to the previous sequence or the writer is already closed of
+     *                                  the sequence has been already added.
+     * @throws IOException              if such exception is thrown when writing into the output resources.
+     */
+    public FastaReferenceWriter startSequence(final String sequenceName, final String description, final int basesPerLine)
+            throws IOException {
+        assertIsNotClosed();
+        checkSequenceName(sequenceName);
+        final String nonNullDescription = checkDescription(description);
+        checkBasesPerLine(basesPerLine);
+        closeSequence();
+        if (sequenceNamesAndSizes.containsKey(sequenceName)) {
+            throw new IllegalStateException("the input sequence name '" + sequenceName + "' has already been added");
+        }
+        currentSequenceName = sequenceName;
+        currentBasesPerLine = basesPerLine;
+        final StringBuilder builder = new StringBuilder(sequenceName.length() + nonNullDescription.length() + 10);
+        builder.append(HEADER_START_CHAR).append(sequenceName);
+        if (!nonNullDescription.isEmpty()) {
```

Comment:
This check is unnecessary, if description is empty then using `append()` will have no effect

Reply:
what about the separator?

Reply:
You're right, I missed the other append.